common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -149,13 +149,38 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     }
   }
 
+  def cleanAmpMaps(document: Document) = {
+    document.getElementsByClass("element-map").foreach { embed: Element =>
+      embed.getElementsByTag("iframe").map { element: Element =>
+        val src = element.attr("src")
+        val frameBorder = element.attr("frameborder")
+        val elementMap = document.createElement("amp-iframe")
+          elementMap
+      // In AMP, when using the layout `responsive`, width is 100%,
+      // and height is decided by the ratio between width and height.
+      // https://www.ampproject.org/docs/guides/responsive/control_layout.html
+            .attr("width", "4")
+            .attr("height", "3")
+            .attr("layout", "responsive")
+            .attr("sandbox", "allow-scripts allow-same-origin allow-popups")
+            .attr("frameborder", frameBorder)
+            .attr("src", src)
+          embed
+            .empty()
```

Comment:
I've had a crack at using replaceWith instead of empty and appendChild. 
The major difference between the two seems to be that replaceWith doesn't return anything but works as a side-effect, whereas appendChild returns an element.
Let me know what you think.